pkg/bootstrap/bootstrap_config_test.go

```diff
@@ -66,18 +82,74 @@ func TestGolden(t *testing.T) {
 				t.Error("Error reading generated file ", err)
 				return
 			}
+
+			// apply minor modifications for the generated file so that tests are consistent
+			// across different env setups
+			err = ioutil.WriteFile(fn, correctForEnvDifference(data), 0700)
+			if err != nil {
+				t.Error("Error modifying generated file ", err)
+				return
+			}
+
+			// re-read generated file with the changes having been made
+			data, err = ioutil.ReadFile(fn)
+			if err != nil {
+				t.Error("Error reading generated file ", err)
+				return
+			}
+
 			golden, err := ioutil.ReadFile("testdata/" + c.base + "_golden.json")
 			if err != nil {
 				golden = []byte{}
 			}
 			if string(data) != string(golden) {
 				t.Error("Generated incorrect config, want:\n" + string(golden) + "\ngot:\n" + string(data))
 			}
+
+			// Check if the LightStep access token file exists
+			_, err = os.Stat(lightstepAccessTokenFile(path.Dir(fn)))
+			if c.expectLightstepAccessToken {
+				if os.IsNotExist(err) {
+					t.Error("expected to find a LightStep access token file but none found")
+				} else if err != nil {
+					t.Error("error running Stat on file: ", err)
+				}
+			} else {
+				if err == nil {
+					t.Error("found a LightStep access token file but none was expected")
+				} else if !os.IsNotExist(err) {
+					t.Error("error running Stat on file: ", err)
+				}
+			}
 		})
 	}
 
 }
 
+type regexReplacement struct {
+	pattern     *regexp.Regexp
+	replacement []byte
+}
+
+// correctForEnvDifference corrects the portions of a generated bootstrap config that vary depending on the environment
+// so that they match the golden file's expected value.
+func correctForEnvDifference(in []byte) []byte {
```

Comment:
The access token is written into a file with that path being put into the envoy bootstrap. Since that path is dependent upon the environment, anytime someone runs tests locally they will end up with a different path which then will cause the golden file to be invalid. This was a workaround which replaces that particular occurrence with a standard path.